chain.go

```diff
@@ -154,13 +154,6 @@ type Chain[B Block] struct {
 }
 
 type ToolsConfig[B Block] struct {
-
-	// SanitizeBlockForCompare is a function that takes a chain agnostic [block] and transforms it in-place, removing fields
-	// that should not be compared.
-	//
-	// The [SanitizeBlockForCompare] is optional, if nil, no-op sanitizer be used.
-	SanitizeBlockForCompare SanitizeBlockForCompareFunc[B]
-
```

Comment:
Don't remove, those are used in `firehose-ethereum` and also battlefield. Please be really mindful of backward compatibility. We are in a version 1.x so we must keep backward compatibility. Same for `GetSanitizeBlockForCompare`

They should be re-wired too so that the functionality works as expected.

I need those to work.